iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -668,3 +714,46 @@ TEST_F(PortManager_test, PortDestroy)
         }
     }
 }
+
+TEST_F(PortManager_test, AcquireInterfacePortDataAfterDestroyingPreviouslyAcquiredOnesIsSuccessful)
+{
+    std::vector<iox::popo::InterfacePortData*> interfaceContainer;
+    std::string itf = "itf";
+
+    // first aquire all possible interfaces
+    exhaustInterfacePorts(itf, [&](auto InterafcePort) { interfaceContainer.push_back(InterafcePort); });
+
+    // so now no one should be available
+    {
+        auto interp = m_portManager->acquireInterfacePortData(iox::capro::Interfaces::INTERNAL, m_Process);
+        EXPECT_EQ(interp, nullptr);
+    }
+
+    // set the destroy flag and let the discovery loop take care
+    setDestroyFlagAndClearContainer(interfaceContainer);
+    m_portManager->doDiscovery();
+
+    // so we should able to get some more now
+    for (uint32_t i = 0U; i < iox::MAX_INTERFACE_NUMBER; i++)
+    {
+        auto interp = m_portManager->acquireInterfacePortData(iox::capro::Interfaces::INTERNAL, m_Process);
+        EXPECT_NE(interp, nullptr);
+    }
+}
+
+TEST_F(PortManager_test, OfferPublisherServiceUpdatesServiceRegistryChangeCounter)
+{
+    auto serviceCounter = m_portManager->serviceRegistryChangeCounter();
+    auto initialCount = serviceCounter->load();
+    PublisherOptions publisherOptions{1};
+
+    PublisherPortUser publisher(
+        m_portManager
+            ->acquirePublisherPortData({1, 1, 1}, publisherOptions, m_Process, m_payloadMemoryManager, PortConfigInfo())
+            .value());
```

Comment:
could you please acquire the publisher port data separately and `ASSERT` that is has no error before using `vlaue()`